src/app/shared/directives/snipcart-add.directive.ts

```diff
@@ -37,17 +41,16 @@ export class SnipcartAddDirective implements OnInit, OnChanges {
     }
 
     private setCustomAttributes(): void {
-        this.modifiers.forEach((option, index) => {
-            const basename = `data-item-custom${ index + 1 }`;
-            const optionList = this.rawListToSnipcartList(option.list);
-
-            this.el.nativeElement.setAttribute(basename + "-name", option.name);
-            this.el.nativeElement.setAttribute(basename + "-options", optionList);
-            this.el.nativeElement.setAttribute(basename + "-value", option.selection);
-        });
-    }
-
-    private rawListToSnipcartList(raw: string[]): string {
-        return raw.reduce((prev, curr) => prev + "|" + curr);
+        this.modifiers
+            .forEach(
+                (modifier, index) => {
+                    const basename = `data-item-custom${ index + 1}`;
+                    const optionsList = this.attributeToSnipcartListPipe.transform(modifier.attribute);
```

Comment:
There is some inconsistency in the variables, methods etc.

You can the variable optionsList, but pass an attribute to the pipe